tokio/src/sync/mpsc/bounded.rs

```diff
@@ -1036,6 +1036,37 @@ impl<T> Sender<T> {
             chan: self.chan.downgrade(),
         }
     }
+
+    /// Returns the max buffer capacity of the channel.
+    ///
+    /// The max capacity is the buffer capacity initially specified when calling
+    /// [`channel`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::sync::mpsc;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (tx, rx) = mpsc::channel::<()>(5);
+    ///      
+    ///     // both max capacity and capacity are the same at first
+    ///     assert_eq!(tx.max_capacity(), 5);
+    ///     assert_eq!(tx.capacity(), 5);
+    ///
+    ///     // Making a reservation doesn't change the max capacity.
+    ///     let permit = tx.reserve().await.unwrap();
+    ///     assert_eq!(tx.max_capacity(), 5);
+    ///     // but drops the capacity by one
+    ///     assert_eq!(tx.capacity(), 4);
+    /// }
```

Comment:
we'll probably need to add
```suggestion
    /// # drop(rx);
    /// }
```
here to suppress unused-variable warnings?

Reply:
I suppose we didn't need to drop since the variable is already prefixed with `_` ?

Reply:
it doesn't look like it is to me? https://github.com/tokio-rs/tokio/blob/a88fb0f1f89ce75abc1b817d034b96546f2f8350/tokio/src/sync/mpsc/bounded.rs#L1060

Reply:
This is fixed